pkg/p2p/kadcast/network.go

```diff
@@ -61,5 +62,59 @@ func sendUDPPacket(netw string, addr net.UDPAddr, payload []byte) {
 	if err != nil {
 		log.WithError(err).Warn("Error while writting to the filedescriptor.")
 		return
-	} 
-}
\ No newline at end of file
+	}
+}
+
+// StartTCPListener listens infinitely for TCP packet arrivals and
+// executes it's processing inside a gorutine by sending
+// the packets to the circularQueue.
+func StartTCPListener(netw string, queue *ring.Buffer, MyPeerInfo Peer) {
+	lAddr := getLocalTCPAddress()
+	// Set listening port.
+	lAddr.Port = int(MyPeerInfo.port)
+PacketConnCreation:
+	// listen to incoming TCP packets
+	listener, err := net.ListenTCP(netw, &lAddr)
+	if err != nil {
+		log.Panic(err)
+	}
+	// Set initial deadline.
+	listener.SetDeadline(time.Now().Add(time.Minute))
+
+	// Instanciate the buffer
+	buffer := make([]byte, 5000000)
+	for {
+		// Read TCP packet.
+		pc, err := listener.AcceptTCP()
+		uAddr := pc.RemoteAddr()
+		byteNum, err := pc.Read(buffer)
```

Comment:
Using the plain `Read` method on a TCP connection never guarantees how many bytes will be read (I've had messages which were 8 bytes long be cut off in the middle due to usage of `Read`). It might be better to refactor this with `io.ReadFull`, and employing some strategy which deduces the size of the packet from some header with a pre-determined length (if this is available in Kadcast, otherwise we can maybe read up to the actual message header).

Reply:
I've thought that since this is `TCP` we are sure that we will get the whole stream of bytes or an err. 
So having that in mind, it's not important at all how many bytes were read.
Will refactor it.

Reply:
It's weird, it works fine for UDP messages. 
Anyway, will implement the `io` buffer then since kadcast does not send this info on the headers and also, it is only used later for encoding the packets correctly through the circular queue.